### PR TITLE
Light fixture code update

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Lights now use overlays for their bulb states.
Fixed up/moved around a whole bunch of code. Light fittings the item (and light bulbs) are in their own dm files.
Cleaned up some proc chain stuff/put things in the right place.

Killed some unneeded stuff

Spark effects actually work again.
Smashing lights as humans works correctly again.

Crucially fixed cigarettes so they can be lit from broken lights, flaming tiles, or flaming mobs (including yourself)

I didn't touch bulbs/fittings other than moving them, so there's still shitcode that could be cleaned up, but its better than before at least.
## Why It's Good For The Game
Better code or something.
## Changelog
:cl:
fix: fixed a couple minor bugs with light fittings
code: Cleaned up some light fitting code
fix: You can once again light cigarettes using broken lights, burning turfs, or burning mobs
/:cl:

### CODE REVIEW

File: code/modules/power/light_bulbs.dm [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	icon = 'icons/obj/lighting.dmi'
+	item_icons = list(
+		slot_l_hand_str = 'icons/mob/inhands/equipment/lights_left.dmi',
+		slot_r_hand_str = 'icons/mob/inhands/equipment/lights_right.dmi',
+	)
+	force = 2
+	throwforce = 5
+	w_class = WEIGHT_CLASS_SMALL
+	var/status = 0		// LIGHT_OK, LIGHT_BURNED or LIGHT_BROKEN
+	var/bulb_type
```
Comment (QualityVan) [inline code/modules/power/light_bulbs.dm:15]: Is this used for anything you couldn't just initial(iconstate) instead?

**Review by QualityVan — CHANGES_REQUESTED** [review]
First pass, will do a longer look at power/lighting later.

File: code/game/objects/items/tools/flame_tools.dm [hunk 2]
```diff
@@ -354,28 +354,32 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-				return
+/obj/item/clothing/mask/cigarette/attack_obj(obj/target_object, mob/living/user)
+	if(lit)
+		return ..()
+	if(!istype(target_object, /obj/machinery/light))
+		return ..()
+	var/obj/machinery/light/fixture = target_object
+	if(fixture.status != LIGHT_BROKEN || !fixture.has_power())
+		return ..()
+	light(span_notice("[user] lights [user.p_their()] [src] from the broken light."))
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/game/objects/items/tools/flame_tools.dm:365]: wdym?

File: code/modules/power/light_bulbs.dm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	var/switchcount = 0	// number of times switched
+	var/rigged = 0		// true if rigged to explode
+	var/brightness = 2 //how much light it gives off
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/modules/power/light_bulbs.dm:18]: done

File: code/modules/power/light_bulbs.dm [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		to_chat(user, span_notice("The replacement bulb is broken."))
+		return
+	var/obj/item/stack/tile/light/existing_bulb = light_tile.floor_tile
+	if(existing_bulb.state == LIGHT_TILE_OK)
+		to_chat(user, span_notice("The lightbulb seems fine, no need to replace it."))
+		return
+
+	user.drop_held_item(src)
+	qdel(src)
+	existing_bulb.state = 0
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/modules/power/light_bulbs.dm]: done

File: code/modules/power/light_bulbs.dm [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	if(istype(target, /obj/machinery/light))
+		return
+	if(user.a_intent != INTENT_HARM)
+		return
+
+	shatter()
+
+/obj/item/light_bulb/proc/shatter()
+	if(status == LIGHT_OK || status == LIGHT_BURNED)
+		src.visible_message(span_warning(" [name] shatters."),span_warning(" You hear a small glass object shatter."))
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/modules/power/light_bulbs.dm]: done

File: code/modules/power/lighting_frame.dm [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+	name = "light fixture frame"
+	desc = "A light fixture under construction."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "tube-construct-stage1"
+	anchored = TRUE
+	layer = FLY_LAYER
+	var/stage = 1
+	var/fixture_type = "tube"
+	var/sheets_refunded = 2
+	var/obj/machinery/light/newlight = null
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/modules/power/lighting_frame.dm]: done

File: code/modules/power/light_bulbs.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/power/light_bulbs.dm]: used base icon state

File: code/modules/power/lighting.dm [hunk 7]
```diff
@@ -385,106 +174,62 @@
...
+	visible_message("[user] smashed the light!", "You hit the light, and it smashes!")
+	if(light_on && (I.atom_flags & CONDUCT) && prob(12))
+		electrocute_mob(user, get_area(src), src, 0.3)
+	broken()
+
+/obj/machinery/light/screwdriver_act(mob/living/user, obj/item/I)
+	if(user.a_intent == INTENT_HARM)
+		return FALSE
+	if(status != LIGHT_EMPTY)
+		balloon_alert(user, "Remove bulb")
```
Comment (QualityVan) [inline code/modules/power/lighting.dm:205]: Was confused for a bit but this is an instruction rather than an action, yeah? Feels unclear to me.

Comment (Lumipharon) [reply code/modules/power/lighting.dm]: updated

**Review by QualityVan — APPROVED** [review]

**Pull request merged**